CMakeLists.txt

```diff
@@ -161,8 +161,9 @@ endif()
 check_clang_version()
 
 if (${CLANG_VERSION_MAJOR} VERSION_GREATER ${CLANG_HIGHEST_VERSION})
```

Comment:
Can you add a TODO here to say that we should merge all the code blocks involving CLANG_EXECUTABLE ?